start_vscode_bessemer.sh

```diff
@@ -0,0 +1,478 @@
+
+# check for reconnect_info in the current directory on the local computer
+echo -e "Checking for left over files from previous sessions"
+if [ -f $VSC_SCRIPTDIR/reconnect_info ]; then
+        echo -e "Found old reconnect_info file, deleting it ..."
+        rm $VSC_SCRIPTDIR/reconnect_info
+fi
+
+# check for log files from a previous session in the home directory of the cluster
+ssh -T $VSC_SSH_OPT <<ENDSSHIP
+if [ -f /home/$VSC_USERNAME/vscip ]; then
+        echo -e "Found old vscip file, deleting it ..."
+        rm /home/$VSC_USERNAME/vscip
+fi
+ENDSSHIP
+
+ssh -T $VSC_SSH_OPT <<ENDSSHPORT
+if [ -f /home/$VSC_USERNAME/vscport ]; then
+        echo -e "Found old vscport file, deleting it ..."
+        rm /home/$VSC_USERNAME/vscport
+fi
+ENDSSHPORT
+
+VSCJIDPRESENT=$(ssh $VSC_SSH_OPT "[ -e ~/vscjid ] && echo 1 || echo 0")
+
+if [[ "$VSCJIDPRESENT" == 1 ]] ; then
+        echo -e "Found old vscjid file, are you already running a session?  Remove /home/$VSC_USERNAME/vscjid if you are sure this is not a duplicate session and re-run script. Exiting!"
+        exit 1
+fi
+
+###############################################################################
+# Check required SSL certs exist                                              #
+###############################################################################
+
+SSLCERT=$(ssh $VSC_SSH_OPT "[ -e ~/.ssl/vscoderemote/vscode_remote_ssl-server-cert.pem ] && echo 1 || echo 0")
+SSLCERTKEY=$(ssh $VSC_SSH_OPT "[ -e ~/.ssl/vscoderemote/private/vscode_remote_ssl-server-key.pem ] && echo 1 || echo 0")
+
+if [[ "$SSLCERT" == 0 ]] || [[ "$SSLCERTKEY" == 0 ]] ; then
+        echo -e "Missing SSL certificate or key. Exiting! Please 'module load apps/vscode-server/4.2.0/binary' run SSL setup step 'setup_ssl_ca_server_client.sh' first! "
+        exit 1
+fi
+
+###############################################################################
+# Start code-server on the cluster                                            #
+###############################################################################
+
+# Make a random password
+VSCPASS=$(strings /dev/urandom | grep -o '[[:alnum:]]' | head -n 30 | tr -d '\n'; echo)
+
+# This is being done this way for reasons.
+ssh -T $VSC_SSH_OPT "sed -i '/^password:/d' ~/.config/code-server/config.yaml && echo 'password: $VSCPASS' > ~/.config/code-server/config.yaml"
+
+# run the code-server job on Bessemer and save the ip of the compute node in the file vscip in the home directory of the user on Bessemer
+echo -e "Connecting to $VSC_HOSTNAME to start the code-server in a batch job"
+# FIXME: save jobid in a variable, that the script can kill the batch job at the end
+echo -e "Connection command:"
+echo -e "==================================================================================="
+echo -e "ssh ${VSC_SSH_OPT} sbatch --export=ALL --cpus-per-task=${VSC_CPUS_PER_TASK} -t=${VSC_RUN_TIME} --mem=${VSC_MEM_PER_JOB}G ${VSC_SNUM_GPU}"
```

Comment:
We may need to add a warning on both modules / scripts that the default behavior is set to inherit the environment from the users' login session (i.e. .bashrc) even though strictly we are doing this intentionally to ensure their personal software  (if they have any)  / environment is also loaded.

Reply:
Resolving this as we can track it in https://github.com/rcgsheffield/vscoderemote_sheffield_hpc/issues/3